src/Microsoft.AspNet.Server.Kestrel/Infrastructure/KestrelTrace.cs

```diff
@@ -1,85 +1,109 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-//using System.Diagnostics.Tracing;
+using System;
+using Microsoft.Framework.Logging;
 
 namespace Microsoft.AspNet.Server.Kestrel
 {
     /// <summary>
     /// Summary description for KestrelTrace
     /// </summary>
-    public class KestrelTrace //: EventSource
+    public class KestrelTrace : ILogger
     {
-        public static KestrelTrace Log = new KestrelTrace();
-  //      static EventTask Connection = (EventTask)1;
-    //    static EventTask Frame = (EventTask)1;
+        private static ILogger _logger;
+        private static readonly KestrelTrace _instance = new KestrelTrace();
 
+        public static void Initialize(ILogger logger)
+        {
+            _logger = logger;
+        }
+
+        private KestrelTrace()
+        {
+        }
+
+        public static KestrelTrace Log
+        {
+            get
+            {
+                if (_logger == null)
+                {
+                    throw new InvalidOperationException($"{nameof(KestrelTrace)} is not initialized. Please call {nameof(Initialize)}({nameof(ILogger)}) method first.");
+                }
+
+                return _instance;
+            }
+        }
 
-      //  [Event(13, Level = EventLevel.Informational, Message = "Id {0}")]
         public void ConnectionStart(long connectionId)
         {
-         //   WriteEvent(13, connectionId);
+            _logger.LogDebug(13, $"{nameof(ConnectionStart)} -> Id: {connectionId}");
         }
 
-      //  [Event(14, Level = EventLevel.Informational, Message = "Id {0}")]
         public void ConnectionStop(long connectionId)
         {
-       //     WriteEvent(14, connectionId);
+            _logger.LogDebug(14, $"{nameof(ConnectionStop)} -> Id: {connectionId}");
         }
 
-
-   //     [Event(4, Message = "Id {0} Status {1}")]
         internal void ConnectionRead(long connectionId, int status)
         {
-     //       WriteEvent(4, connectionId, status);
+            _logger.LogDebug(4, $"{nameof(ConnectionRead)} -> Id: {connectionId}, Status: {status}");
```

Comment:
This might be a little verbose, even for debug logging.

@loudej Do you have any opinions on this? Should we just not log this at all by default? Is there anything else we shouldn't log by default like like ConnectionReadStart?